src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/ExceptionsController.cs

```diff
@@ -21,43 +23,50 @@ namespace Microsoft.Diagnostics.Monitoring.WebApi.Controllers
     [Authorize(Policy = AuthConstants.PolicyName)]
     [ProducesErrorResponseType(typeof(ValidationProblemDetails))]
     [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
-    public sealed class ExceptionsController : ControllerBase
+    public sealed class ExceptionsController :
+        DiagnosticsControllerBase
     {
-        private readonly IExceptionsStore _exceptionsStore;
-        private readonly IOptions<ExceptionsOptions> _exceptionsOptions;
-        private readonly IExceptionsOperationFactory _operationFactory;
+        private readonly IOptions<ExceptionsOptions> _options;
 
-        public ExceptionsController(IServiceProvider serviceProvider)
+        public ExceptionsController(
+            IServiceProvider serviceProvider,
+            ILogger<ExceptionsController> logger)
+            : base(serviceProvider.GetRequiredService<IDiagnosticServices>(), logger)
         {
-            // The exceptions store for the default process
-            _exceptionsStore = serviceProvider.GetRequiredService<IExceptionsStore>();
-            _exceptionsOptions = serviceProvider.GetRequiredService<IOptions<ExceptionsOptions>>();
-            _operationFactory = serviceProvider.GetRequiredService<IExceptionsOperationFactory>();
+            _options = serviceProvider.GetRequiredService<IOptions<ExceptionsOptions>>();
         }
 
         /// <summary>
-        /// Gets the exceptions from the default process.
+        /// Gets the exceptions from the target process.
         /// </summary>
         [HttpGet("exceptions", Name = nameof(GetExceptions))]
         [ProducesWithProblemDetails(ContentTypes.ApplicationNdJson, ContentTypes.ApplicationJsonSequence, ContentTypes.TextPlain)]
         [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
-        [EgressValidation]
-        public ActionResult GetExceptions()
+        public Task<ActionResult> GetExceptions(
+            [FromQuery]
+            int? pid = null,
+            [FromQuery]
+            Guid? uid = null,
+            [FromQuery]
+            string name = null)
         {
-            if (!_exceptionsOptions.Value.GetEnabled())
+            if (!_options.Value.GetEnabled())
             {
-                return NotFound();
+                return Task.FromResult<ActionResult>(NotFound());
             }
 
-            ExceptionsFormat? format = ComputeFormat(Request.GetTypedHeaders().Accept);
-            if (!format.HasValue)
+            ProcessKey? processKey = Utilities.GetProcessKey(pid, uid, name);
+
+            return InvokeForProcess(processInfo =>
             {
-                return this.NotAcceptable();
-            }
+                ExceptionsFormat format = ComputeFormat(Request.GetTypedHeaders().Accept) ?? ExceptionsFormat.PlainText;
```

Comment:
Is this to bring the default behavior in-line with what stacks does? No issue with the change, just wanted to know the motivation.

Reply:
Yes, we should provide a default content type as a response if the client doesn't ask for one.